tests/core/full_node/test_mempool.py

```diff
@@ -1878,3 +1932,203 @@ def test_parse_unknown_condition(self):
 
             with pytest.raises(ValidationError):
                 cost, args = parse_condition_args(SExp.to([]), opcode, False)
+
+
+# the following tests generate generator programs and run them through get_name_puzzle_conditions()
+
+COST_PER_BYTE = 12000
+MAX_BLOCK_COST_CLVM = 11000000000
+
+
+def generator_condition_tester(
+    conditions: str, safe_mode: bool = False, quote: bool = True, max_cost: int = MAX_BLOCK_COST_CLVM
+) -> NPCResult:
+    prg = f"(q ((0x0101010101010101010101010101010101010101010101010101010101010101 {'(q ' if quote else ''} {conditions} {')' if quote else ''} 123 (() (q . ())))))"  # noqa
+    print(f"program: {prg}")
+    program = SerializedProgram.from_bytes(binutils.assemble(prg).as_bin())
+    generator = BlockGenerator(program, [])
+    print(f"len: {len(bytes(program))}")
+    npc_result: NPCResult = get_name_puzzle_conditions(
+        generator, max_cost, cost_per_byte=COST_PER_BYTE, safe_mode=safe_mode
+    )
+    return npc_result
+
+
+class TestGeneratorConditions:
+    def test_duplicate_height_time_conditions(self):
+        # ASSERT_SECONDS_RELATIVE
+        # ASSERT_SECONDS_ABSOLUTE
+        # ASSERT_HEIGHT_RELATIVE
+        # ASSERT_HEIGHT_ABSOLUTE
+        for cond in [80, 81, 82, 83]:
+            # even though the generator outputs multiple conditions, we only
+            # need to return the highest one (i.e. most strict)
+            npc_result = generator_condition_tester(" ".join([f"({cond} {i})" for i in range(50, 101)]))
+            assert npc_result.error is None
+            assert len(npc_result.npc_list) == 1
+            opcode = ConditionOpcode(bytes([cond]))
+            max_arg = 0
+            assert npc_result.npc_list[0].conditions[0][0] == opcode
+            for c in npc_result.npc_list[0].conditions[0][1]:
+                assert c.opcode == opcode
+                max_arg = max(max_arg, int_from_bytes(c.vars[0]))
+            assert max_arg == 100
+
+    def test_just_announcement(self):
+        # CREATE_COIN_ANNOUNCEMENT
+        # CREATE_PUZZLE_ANNOUNCEMENT
+        for cond in [60, 62]:
+            message = "a" * 1024
+            # announcements are validated on the Rust side and never returned
+            # back. They ar either satisified or cause an immediate failure
+            npc_result = generator_condition_tester(f'({cond} "{message}") ' * 50)
+            assert npc_result.error is None
+            assert len(npc_result.npc_list) == 1
+            # create-announcements and assert-announcements are dropped once
+            # validated
+
+    #            assert npc_result.npc_list[0].conditions == []
```

Comment:
these odd commented out checks are meant to be enabled once more validation happens in the conditions parser (i.e. in rust)